mlflow/server/js/src/components/ExperimentView.js

```diff
@@ -108,6 +109,8 @@ class ExperimentView extends Component {
       showDeleteRunModal: false,
       // True if a model for restoring one or more runs should be displayed
       showRestoreRunModal: false,
+      // Last index of a clicked run checkbox, within a list of runs sorted by display order
+      lastCheckboxIndex: undefined,
```

Comment:
Btw, some quirks:
- this seems to order child runs "backwards". If I select the top-most child run, and then shift-select a few parent runs down, the selection skips over the lower children runs. If I select the bottom-most child run, though, it will select everything.
- Clicking the "select all" checkbox does not clear the last selected checkbox, nor does changing the sort order. In fact it seems like most other operations should clear the last selected checkbox. Here we see some complexities: Gmail maintains a "cursor" which makes it obvious where the current shift would start from. GitHub, on the other hand, requires that you start holding shift prior to making your selection, allowing it to be stateless.

Reply:
Good catch, this should be fixed now - the earlier logic didn't take into account the sorting order of child runs during shift-selection